Saul Eduardo Diaz Fonseca
<br>
19/08/2026

## ACTIVIDAD 1
<br>
¿Quién produjo originalmente los datos?
<br>
 World Population Prospects
 <br>
¿Qué organización los procesa para su utilización en Our World in Data?
<br>
Our World in Data
<br>
¿Qué representa la variable que analizarás?
<br>
Crecimiento anual de población en los países.


## ACTIVIDAD 2

In [17]:
import pandas as pd #Leer datos
import numpy as np 

In [18]:
dataframe = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

In [ ]:
#Obtener primeras 5 filas
dataframe.head(10)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN
5,Afghanistan,AFG,1956,127322.0,NaN
6,Afghanistan,AFG,1957,134037.0,NaN
7,Afghanistan,AFG,1958,135072.0,NaN
8,Afghanistan,AFG,1959,145858.0,NaN
9,Afghanistan,AFG,1960,165778.0,NaN


In [ ]:
#Obtener dimensiones de tabla ( filas , columnas )
dataframe.shape

(38400, 5)

In [21]:
#Nombre de columnas
dataframe.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [22]:
#Tipo de datos
dataframe.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [23]:
#Resumen estadistico
dataframe.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [24]:
#Cantidad de valores faltantes
dataframe.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [25]:
#Cantidad de entidades ( países )
dataframe["Entity"].nunique()

256

 ## PREGUNTAS 

#### ¿Cuántas filas y columnas tiene el dataset?

38500 columnas y 5 filas

#### ¿Qué tipo de datos contiene la variable Year?
int64

#### ¿Existen valores faltantes?

Si, Code, Anual change un population y annual population change projected.

#### ¿Cuántas entidades diferentes contiene el dataset?

256 entidades ( países )


## ACTIVIDAD 3 

In [26]:
#Media ( 2020 )
df_2020 = dataframe[dataframe['Year'] == 2020]
poblacion_2020 = df_2020['Annual change in population']

#identificar minimos y maximos

entidad_min = df_2020.loc[poblacion_2020.idxmin()]["Entity"]
entidad_max = df_2020.loc[poblacion_2020.idxmax()]["Entity"]

print ("Media: ", poblacion_2020.mean() )
print("Mediana :",poblacion_2020.median())
print("Varianza :",poblacion_2020.var())
print("Desviacion estandar :",poblacion_2020.std())
print("Minimo :" , poblacion_2020.min() , "en" , entidad_min)
print("Maximo :",poblacion_2020.max() , "en" , entidad_max)

Media:  2124908.84765625
Mediana : 34293.5
Varianza : 90687335031103.64
Desviacion estandar : 9522989.815761836
Minimo : -494016.0 en Venezuela
Maximo : 75707590.0 en World


### PREGUNTAS

#### ¿Cuál es el valor de la media?
2124908.84765625
#### ¿Cuál es el valor de la mediana?
34293.5
#### ¿Cuál es el valor de la desviación estándar?
9522989.815761836
#### ¿Qué entidad presenta el valor mínimo y cuál el máximo?
Mínimo en Venezuela y máximo en todo el mundo.


## ACTIVIDAD 4



In [ ]:
#Mostrar versiones
import sys
import notebook

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)
print("Jupyter Notebook:", notebook.__version__)

Python: 3.13.6 (tags/v3.13.6:4e66535, Aug  6 2025, 14:36:00) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2
Jupyter Notebook: 7.6.2


In [ ]:
#SI se usa un solo corchete me trae la serie ( indice - valor) ( dataframe[year])
#SI se usa dos, te trae la información con formato dataframe ( tabla ) ( dataframe [[year]])
df_year = dataframe[["Year","Entity"]]
df_year.head()

,Year,Entity
0,1951,Afghanistan
1,1952,Afghanistan
2,1953,Afghanistan
3,1954,Afghanistan
4,1955,Afghanistan


In [29]:
#Filtro para filas ( todas donde tenga year 2020)
df_extraccion_filas = dataframe[ dataframe["Year"] == 2020]
df_extraccion_filas.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
69,Afghanistan,AFG,2020,1212852.0,NaN
219,Africa (UN),UN_AFR,2020,32815675.0,NaN
369,Albania,ALB,2020,-13061.0,NaN
519,Algeria,DZA,2020,747543.0,NaN
669,American Samoa,ASM,2020,-451.0,NaN


In [30]:
#Poner varias condiciones para filas
df_2020_mexico = dataframe[(dataframe["Entity"] == "Mexico") & ( dataframe["Year"].between(2020,2021))] [["Entity","Year","Annual change in population"]]
df_2020_mexico.head()

,Entity,Year,Annual change in population
22269,Mexico,2020,1036075.0
22270,Mexico,2021,849098.0


In [31]:
# Sacar medidas con diferentes tipos de datos
df_2020_mexico.mean(numeric_only= True) # usar numeric para solo sacar media de valores numericos.

# En promedio en un rango desde el 2006 y 2020 el cambio de la poblacion es de 1,399,170

Year                             2020.5
Annual change in population    942586.5
dtype: float64

In [123]:
# Localizar por valor del indice LOC (fila columna)
dataframe.loc[:,["Entity","Year"]]

,Entity,Year
0,Afghanistan,1951
1,Afghanistan,1952
2,Afghanistan,1953
3,Afghanistan,1954
4,Afghanistan,1955
...,...,...
38395,Zimbabwe,2096
38396,Zimbabwe,2097
38397,Zimbabwe,2098
38398,Zimbabwe,2099


In [45]:
#Localizar por posicion del indice ( solo acepta numeros )
dataframe.iloc[0:3,0:2]

,Entity,Code
0,Afghanistan,AFG
1,Afghanistan,AFG
2,Afghanistan,AFG


In [ ]:
# Traer posicion 10,000
dataframe.loc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

In [61]:
dataframe.loc[:,["Entity","Year","Annual change in population"]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [64]:
#Combinar ambos
dataframe.loc[10000,["Entity","Year","Annual change in population"]]

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

In [91]:
#4
dataframe.loc[(dataframe["Year"] == 2020) & (dataframe["Entity"] == "Mexico")]


,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [92]:

#5
dataframe.iloc[0:6]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN
5,Afghanistan,AFG,1956,127322.0,NaN


In [93]:
#6
dataframe.iloc[:,[0,2,3]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [ ]:
#7
dataframe.iloc[0:4,0:3]

[        Entity Code  Year
0  Afghanistan  AFG  1951
1  Afghanistan  AFG  1952
2  Afghanistan  AFG  1953
3  Afghanistan  AFG  1954]


In [ ]:
# Año 2005, letra apellido
data = dataframe.loc[(dataframe["Year"] >= 2006) & 
              (dataframe["Year"] <= 2020) &
              (dataframe["Entity"] == "Switzerland")
            ]
data

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
33355,Switzerland,CHE,2006,46690.0,NaN
33356,Switzerland,CHE,2007,66872.0,NaN
33357,Switzerland,CHE,2008,96344.0,NaN
33358,Switzerland,CHE,2009,95914.0,NaN
33359,Switzerland,CHE,2010,88106.0,NaN
33360,Switzerland,CHE,2011,88629.0,NaN
33361,Switzerland,CHE,2012,84548.0,NaN
33362,Switzerland,CHE,2013,92272.0,NaN
33363,Switzerland,CHE,2014,99039.0,NaN
33364,Switzerland,CHE,2015,93702.0,NaN


In [120]:
#Estadisticos
dataNew = data.loc[:,"Annual change in population"]

print ("Media: ",dataNew.mean())
print("Mediana :",dataNew.median())
print("Varianza :",dataNew.var())
print("Desviacion estandar :",dataNew.std())
print("Minimo :" ,dataNew.min())
print("Maximo :",dataNew.max())


Media:  80583.86666666667
Mediana : 88106.0
Varianza : 265762480.4095238
Desviacion estandar : 16302.223173835027
Minimo : 46690.0
Maximo : 99039.0
